tests/CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ target_include_directories(
     INTERFACE
         ${PROJECT_SOURCE_DIR}
         ${CMAKE_CURRENT_SOURCE_DIR}
+        "$<TARGET_PROPERTY:umd::device,INCLUDE_DIRECTORIES>"
```

Comment:
So the line above in target_link_libraries which references umd::device, won't include umd::device's include_directories automatically?

Reply:
When `target_link_libraries` connects up A and B, the `PUBLIC` stuff is brought across.  So all of umd::device's public include directories (and any public compile definitions, dependencies, etc).  But none of its private stuff.

HOWEVER, for tests we have legitimate claim to snoop on the UUT's private bits such as header files.  So here we're sniffing out its `INCLUDE_DIRECTORIES` which are all dirs listed as either `PUBLIC` or `PRIVATE` (only `INTERFACE` ones won't show up here, as those are exported but not used internally).

The mapping is:
[interface, public] -> INTERFACE_INCLUDE_DIRECTORIES
[public, private] -> INCLUDE_DIRECTORIES